src/engine/source/hlp/src/specificParsers.cpp

```diff
@@ -309,18 +370,68 @@ bool parseJson(const char** it, Parser const& parser, ParseResult& result)
 {
     rapidjson::Reader reader;
     rapidjson::StringStream ss {*it};
-    auto h = rapidjson::BaseReaderHandler {};
-    if (reader.Parse<rapidjson::kParseStopWhenDoneFlag>(ss, h).IsError())
+    rapidjson::Document doc;
+
+    // Parse the json and stop at the end of the json object, if error return false
+    // TODO: see if there is a way to specify the root JSON type
+    doc.ParseStream<rapidjson::kParseStopWhenDoneFlag>(ss);
+    if (doc.HasParseError())
     {
         return false;
     }
 
-    hlp::JsonString json {{*it, ss.Tell()}};
-    *it += ss.Tell();
+    // If no errors assert root is the correct type
+    bool valid = false;
+    switch (jsonTypes.at(parser.options[0]))
+    {
+        case JSON_TYPE::J_ANY: valid = true; break;
+        case JSON_TYPE::J_STRING:
+            if (doc.IsString())
+            {
+                valid = true;
+            }
+            break;
+        case JSON_TYPE::J_BOOL:
+            if (doc.IsBool())
+            {
+                valid = true;
+            }
+            break;
+        case JSON_TYPE::J_NUMBER:
+            if (doc.IsNumber())
+            {
+                valid = true;
+            }
+            break;
+        case JSON_TYPE::J_OBJECT:
+            if (doc.IsObject())
+            {
+                valid = true;
+            }
+            break;
+        case JSON_TYPE::J_ARRAY:
+            if (doc.IsArray())
+            {
+                valid = true;
+            }
+            break;
+        case JSON_TYPE::J_NULL:
+            if (doc.IsNull())
+            {
+                valid = true;
+            }
+            break;
+    }
```

Comment:
If the if condition is true, then `valid` is true, so then: 
```suggestion

switch (jsonTypes.at(parser.options[0]))
    {
        case JSON_TYPE::J_ANY: valid = true; break;
        case JSON_TYPE::J_STRING: valid = doc.IsString(); break;
        case JSON_TYPE::J_BOOL: valid = doc.IsBool(); break;
        case JSON_TYPE::J_NUMBER: valid = doc.IsNumber(); break;
        case JSON_TYPE::J_OBJECT: valid = doc.IsObject(); break;
        case JSON_TYPE::J_ARRAY: valid = doc.IsArray(); break;
        case JSON_TYPE::J_NULL: valid = doc.IsNull(); break;
    }
```

Reply:
Nice!